### PR TITLE
rename `UnitQuaternion` to `QuatRotation`

### PR DESCRIPTION
**Pull request by hyrodium**
As discussed in https://github.com/JuliaGeometry/Rotations.jl/issues/171#issuecomment-950391326, `UnitQuaternion` is confusing, and should be renamed to `QuatRotation`.

### CODE REVIEW

File: src/deprecated.jl [hunk 1]
```diff
@@ -2,8 +2,11 @@
 # Deprecate RodriguesVec => RotationVec
 Base.@deprecate_binding RodriguesVec RotationVec true
 
-# Deprecate Quat => UnitQuaternion
-Base.@deprecate_binding Quat UnitQuaternion true
+# Deprecate Quat => QuatRotation
+Base.@deprecate_binding Quat QuatRotation true
```
Comment (c42f) [inline src/deprecated.jl:6]: You can probably remove this old deprecation.

**Review by c42f — APPROVED** [review]
Looks good to me.

File: src/deprecated.jl (hunk 1 above)
Comment (hyrodium) [reply src/deprecated.jl:6]: The old deprecations for `RodriguesVec`, `SPQuat` and `Quat` has been introduced in this commit (#116):
https://github.com/JuliaGeometry/Rotations.jl/pull/116/commits/793d384d14e688887b02914e2cad9fdfbdfbdb65#diff-0c3a3096ea8aa5548f7facdd7eaa649293a64f785376cca25e23f14d6af4301d

If we can remove the deprecation for `Quat`, then I think the others also can be removed.
The old type names were already deprecated before `v1.0.0` release, so I thought it's okay to remove them.

Comment (c42f) [reply src/deprecated.jl:6]: Sure, sounds fine.

**Pull request merged**